### PR TITLE
Matrix free mixed mesh example

### PR DESCRIPTION
**Pull request by dominiktassilostill**
Example for #685

### CODE REVIEW

**Comment by nfehn** [comment]
can you run the indentation script?

I would then just merge this PR for now. (I do not see that we have to discuss the code for now or bring it into any particular shape given that it is a prototype.)

**Comment by dominiktassilostill** [comment]
Sure, also added a function to create a mixed mesh in 3D with a transition from a hypercube area to a simplex area using pyramids.

**Review by kronbichler — APPROVED** [review]

**Comment by nfehn** [comment]
Indentation still seems to fail?

File: prototypes/mixed_meshes.cc [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+#include <fstream>
+
+/**
+ * Adopted from dealii/tests/simplex/matrix_free_02
+ *
+ * Works in 2D with p = 1,2,3
+ * Quads and triangles use the same Gauss quadrature on the faces (which is in 2D lines)
+ *
+ * Does not work in 3D out of the box, need to adopt hp_vertex_dof_identities() and
+ * hp_quad_dof_identities() for FE_Q and FE_SimplexP elements.
```
Comment (peterrum) [inline prototypes/mixed_meshes.cc:52]: https://github.com/dealii/dealii/pull/14607

**Comment by nfehn** [comment]
I guess the new files are not detected automatically by the indentation script and one has to run again e.g. `./config_exadg.sh`?

**Pull request merged**